components/pefa.py

```diff
@@ -0,0 +1,268 @@
+from collections import OrderedDict
+from plotly.subplots import make_subplots
+import numpy as np
+import plotly.graph_objects as go
+import textwrap
+
+SCORE_MAPPING = OrderedDict([
+    (4, "A"),
+    (3, "B"), (3.5, "B+"),
+    (2, "C"), (2.5, "C+"),
+    (1, "D"), (1.5, "D+"),
+])
+
+PILLAR_MAPPING = OrderedDict([
```

Comment:
In the narrative it reads better to have something like "the strongest pillar of Bangladesh’s budget institutions is "Policy-based budgeting" rather than "the strongest pillar of Bangladesh’s budget institutions is "4. Policy-based budgeting".

So it might make sense to remove the leading numbers in the name of the PEFA categories, unless the audience would know and find it helpful to know which numbered category it is.

Reply:
I'm on the fence about this one as the leading number makes it easy to identify the corresponding pillar in the heatmap quickly, but I removed them because you are right, it reads better this way.